source/elements/oneMKL/source/domains/spblas/data_types/init_coo_matrix.rst

```diff
@@ -0,0 +1,251 @@
+
+   nnz
+      The number of explicit entries, also known as Number of Non-Zero elements.
+      Must be at least 0.
+
+   index
+      Indicates how input arrays are indexed. The possible options are described
+      in :ref:`onemkl_enum_index_base` enum class.
+
+   row_ind
+      Buffer of length at least ``nnz`` containing the row indices in
+      ``index``-based numbering. Refer to :ref:`onemkl_sparse_coo` format for
+      detailed description of ``row_ind``.
+
+   col_ind
+      Buffer of length at least ``nnz`` containing the column indices in
+      ``index``-based numbering. Refer to :ref:`onemkl_sparse_coo` format for
+      detailed description of ``col_ind``.
+
+   val
+      Buffer of length at least ``nnz``. Contains the data of the input matrix
+      which is not implicitly zero. The remaining input values are implicit
+      zeros. Refer to :ref:`onemkl_sparse_coo` format for detailed description
+      of ``val``.
+
+.. container:: section
+
+   .. rubric:: Output parameters
+
+   p_smhandle
+      On return, the address is updated to point to a newly allocated and
+      initialized ``matrix_handle_t`` object that can be filled and used to
+      perform sparse BLAS operations.
+
+.. container:: section
+
+   .. rubric:: Notes
+
+   - The parameters ``num_rows``, ``num_cols`` and ``nnz`` may be zero if and
+     only if ``row_ind``, ``col_ind`` and ``val`` are zero-sized, otherwise they
+     must be strictly greater than zero.
+
+.. container:: section
+
+   .. rubric:: Throws
+
+   This function shall throw the following exceptions if the associated
+   condition is detected. An implementation may throw additional
+   implementation-specific exception(s) in case of error conditions not covered
+   here.
+
+   | :ref:`oneapi::mkl::host_bad_alloc<onemkl_exception_host_bad_alloc>`
+   | :ref:`oneapi::mkl::invalid_argument<onemkl_exception_invalid_argument>`
+   | :ref:`oneapi::mkl::unimplemented<onemkl_exception_unimplemented>`
+   | :ref:`oneapi::mkl::unsupported_device<onemkl_exception_unsupported_device>`
+
+.. _onemkl_sparse_init_coo_matrix_usm:
+
+init_coo_matrix (USM version)
+-----------------------------
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       template <typename dataType, typename indexType>
+       void init_coo_matrix (sycl::queue                          &queue,
+                             oneapi::mkl::sparse::matrix_handle_t *p_smhandle,
+                             std::int64_t                         num_rows,
+                             std::int64_t                         num_cols,
+                             std::int64_t                         nnz,
+                             index_base                           index,
+                             indexType                            *row_ind,
+                             indexType                            *col_ind,
+                             dataType                             *val);
```

Comment:
question:  I see that this has a queue provided, so that it is possible for the internals to submit some sort of operation on the data like read ...  but there are no events in or out, so this is assumed to be synchronous and blocking.  For future cases where a newly allocated array may be added to the handle like in SpGEMM operations, this API seems to no longer be functional for keeping an asynchronous cases and it seems we will need to add in separate APIs for updating a matrix handle that can keep the DAG going,  is that our plan ?